modules/standard/Lists.chpl

```diff
@@ -0,0 +1,868 @@
+      //
+      if _blocks == nil then {
+        _sanity(_totalCapacity == 0);
+        _sanity(_size == 0);
+        _blocks = _makeBlockArray(_initialBlockCapacity);
+        _blocks[1] = new unmanaged ListBlock(eltType, _initialCapacity);
+        _totalCapacity += _initialCapacity;
+      }
+
+      const remaining = _totalCapacity - _size;
+      if remaining > amount then
+        return;
+
+      var lastBlockIdx = _getBlockIdx(_size);
+      var req = amount - remaining;
+
+      while req > 0 do {
+        _maybeResizeBlockArray();
+
+        ref oblock = _blocks[lastBlockIdx];
+        lastBlockIdx += 1;
+        ref nblock = _blocks[lastBlockIdx];
+        _sanity(nblock == nil);
+        _sanity(oblock != nil);
+
+        nblock = new unmanaged ListBlock(eltType, oblock.capacity * 2);
+        _totalCapacity += nblock.capacity;
+        req -= nblock.capacity;
+      }
+    }
+
+    pragma "no doc"
+    proc _maybeReleaseMem(in amount: int) {
+      if _blocks == nil || _totalCapacity == 0 then
+        return;
+
+      //
+      // TODO: Worry about writing this _after_ we get the test functions to
+      // work once again.
+      //
+      
+    }
+
+    //
+    // Shift elements including and after index one to the right in memory,
+    // possibly resizing.
+    //
+    pragma "no doc"
+    proc _expand(idx: int) {
+      _sanity(_withinBounds(idx));
+
+      _maybeAcquireMem(1);
+
+      for i in idx.._size by -1 do {
+        ref src = _getRef(i);
+        ref dst = _getRef(i + 1);
+        _move(src, dst);
+      }
+    }
+
+    //
+    // Shift all elements after index "idx" one to the left in memory,
+    // possibly resizing.
+    //
+    pragma "no doc"
+    proc _collapse(idx: int) {
+      _sanity(_withinBounds(idx));
+
+      if idx == _size then
+        return;
+
+      for i in idx.._size do {
+        ref src = _getRef(i + 1);
+        ref dst = _getRef(i);
+        _move(src, dst);
+      }
+
+      _maybeReleaseMem(1);
+    }
+
+    //
+    // Assumes that a copy of the input element has already been made at some
+    // previous boundary, IE giving append's parameter the "in" intent.
+    //
+    pragma "no doc"
+    proc _append(ref x: eltType) {
+      _maybeAcquireMem(1);
+      ref src = x;
+      ref dst = _getRef(_size + 1);
+      _move(src, dst);
+      _size += 1;
+    }
+
+    /*
+      Add an element to the end of this List.
+
+      :arg x: An element to append.
+    */
+    proc append(pragma "no auto destroy" in x: eltType) {
+      _enter();
+      _append(x);
+      _leave();
+    }
+
+    pragma "no doc"
+    inline proc _extendGeneric(collection) {
+
+      //
+      // TODO: This could be faster if we resized once and then performed
+      // repeated moves, rather than calling _append().
+      //
+      for x in collection do
+        _append(x);
+    }
+
+    //
+    // Can't use this for now without getting warnings, so stick with the
+    // simple extend() implementation.
+    //
```

Comment:
If it's not used, maybe comment it out and say something like "We'd like an extendGeneric to be implemented like this ?"